### PR TITLE
HCAP-1492 - add MHAW track to EOI form

### PR DESCRIPTION
**Pull request by coryatwater-FW**
### Pending
- need link for `MENTAL_HEALTH_AND_ADDICTIONS_WORKER_LINK`, it's currently empty, left `TODO` in links.js

### Summary
- added Mental Health and Addictions Worker track to EOI form
- add cypress test for 2 new `conditional` fields
- extract each section to own component to reduce duplicity errors

### New fields
- program
- educationalRequirements
- indigenous
- driverLicense
- experienceWithMentalHealthOrSubstanceUse - (conditional)
- currentOrMostRecentIndustry
- roleInvolvesMentalHealthOrSubstanceUse - (conditional)
- otherIndustry

### JIRA Ticket
[HCAP-1492](https://eydscanada.atlassian.net/browse/HCAP-1492)

### CODE REVIEW

File: client/src/components/participant-form/index.js [hunk 1]
```diff
@@ -35,15 +35,23 @@ export const Form = ({
...
         postalCode: '',
+        indigenous: '',
         preferredLocation: [],
+        driverLicense: '',
+        experienceWithMentalHealthOrSubstanceUse: '',
         consent: '',
         reasonForFindingOut: [],
+        currentOrMostRecentIndustry: '',
+        roleInvolvesMentalHealthOrSubstanceUse: '',
+        otherIndustry: '',
```
Comment (sunghwan-park-fw) [inline client/src/components/participant-form/index.js:54]: Isn't it possible to merge this into currentOrMostRecentIndustry?

Comment (coryatwater-FW) [reply client/src/components/participant-form/index.js:54]: I do have code to do this saved locally, but I wanted to get some opinions! (also this other option may even be removed)

my thought  was having both would make it more apparent of the users `Other` industry as opposed to just having the specified value in `currentOrMostRecentIndustry`.  That way we know the user had selected Other + there industry.

Probably a stretch scenario, but if we ever wanted to sort by industry for example

Comment (sunghwan-park-fw) [reply client/src/components/participant-form/index.js:54]: There might be pros and cons, but the mismatch of questions and fields might cause more trouble than we can expect now.

File: server/validators/helpers.ts [hunk 2]
```diff
@@ -93,15 +95,19 @@ export const errorMessage = ({ path }) => {
 
     // Participant info
     eligibility:
-      "We're sorry, but current eligibility to work in Canada is a requirement to submit this form.",
-    preferredLocation: "Please select at least one location you'd like to work in.",
-    consent: "We're sorry, but we cannot process your request without permission.",
+      "We're sorry, but current eligibility to work in Canada is a requirement to submit this form",
+    educationalRequirements: `We're sorry, but this question is a requirement to submit this form`,
+    preferredLocation: "Please select at least one location you'd like to work in",
+    driverLicense: `We're sorry, but this question is a requirement to submit this form`,
```
Comment (fw-noel) [inline server/validators/helpers.ts:101]: I'd remove We're sorry out of this one, unless you got that from the PEOI? Same with the education requirements one

File: client/src/components/participant-form/index.js (hunk 1 above)
Comment (fw-noel) [reply client/src/components/participant-form/index.js:54]: I agree with Sunghwan that I don't see a huge value add of putting this in a different field

**Review by sunghwan-park-fw — APPROVED** [review]

**Review by sunghwan-park-fw — DISMISSED** [review]

**Review by sunghwan-park-fw — APPROVED** [review]

**Pull request merged**